sway/commands/resize.c

```diff
@@ -256,9 +256,21 @@ static struct cmd_results *resize_adjust_tiled(uint32_t axis,
 		float pct = amount->amount / 100.0f;
 
 		if (is_horizontal(axis)) {
-			amount->amount = (float)current->pending.width * pct;
+            struct sway_container *parent = current->pending.parent;
```

Comment:
This logic is still incorrect, you will need to find the first parallel parent. There is a function `container_find_resize_parent` that does this already that is called by `container_resize_tiled`. Ideally we would need to call it only once to determine the parent.

Reply:
When I try to find the parent with `container_find_resize_parent(current, axis)` call it returns the same `current` pointer back (even if parent exists, idk why yet) and resizing does not work as it should. So, I think I need to check if `parent == current` and if they are equal I should find parent manually again (`parent = current->pending.parent`). I also realized the `container_resize_tiled` function that you mentioned is called at the end of this function that I am working on.